clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3211,7 +3211,12 @@ static void handleWorkGroupSizeHint(Sema &S, Decl *D, const ParsedAttr &AL) {
 
 // Handles intel_reqd_sub_group_size.
 static void handleSubGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
-  if (S.LangOpts.SYCLIsHost)
+  // In SYCLIsHost mode, we want to ignore the attribute. In OpenCL or
+  // SYCLIsDevice mode, we want to process the attribute as normal. This cannot
+  // be handled in Attr.td unless ClangAttrEmitter.cpp gets smarter logic about
+  // when to suppress the "ignored attribute" warnings with a list of language
+  // options.
+  if (!S.LangOpts.SYCLIsDevice && !S.LangOpts.OpenCL)
     return;
```

Comment:
Inside SemaDeclAttr.cpp file, i just noticed that we have
if (!(D->hasAttr<OpenCLKernelAttr>() ||
        LangOpts.SYCLIsDevice || LangOpts.SYCLIsHost)) {
    // These attributes cannot be applied to a non-kernel function.
...
} else if (const auto *A = D->getAttr<IntelReqdSubGroupSizeAttr>()) {
      Diag(D->getLocation(), diag::err_opencl_kernel_attr) << A;
      D->setInvalidDecl();

I think we do not need this checking anymore. @AaronBallman. please correct me if i understand wrong

Reply:
I think we still need that case because the declaration could have the `OpenCLKernelAttr` and the `IntelReqdSubGroupSizeAttr` in OpenCL mode.

Reply:
Thanks for the clarification.